### PR TITLE
Add ability to collect config files with ansible

### PR DESCRIPTION
**Pull request by AstroProfundis**
- Add arguments to manually select config files to collect (similar to #11 )
 - Add funtion to find a process's PID by its listening port (will be used in future features)
 - Minor fixes

### CODE REVIEW

**Comment by AstroProfundis** [comment]
@ethercflow @LinuxGit PTAL

File: measurement/util.py [hunk 1]
```diff
@@ -87,6 +90,17 @@ def parse_insight_opts():
 
     parser.add_argument("--config-file", action="store_true", default=False,
                         help="Collect various configuration files in output, disabled by default.")
+    parser.add_argument("--config-auto", action="store_true", default=False,
+                        help="Automatically detect and save configuration files for all running PD/TiDB/TiKV processes.")
+    parser.add_argument("--config-sysctl", action="store_true", default=False,
+                        help="Save kernel config by collecting output of `sysctl -a`.")
+    parser.add_argument("--config-auto", action="store_true", default=False,
```
Comment (LinuxGit) [inline measurement/util.py]: `--config-auto` argument duplicated

File: measurement/process/meta.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        result = {}
+        for entry in os.scandir("/proc"):
+            # find all PIDs
+            if str.isdigit(entry.name):
+                try:
+                    for _fd in os.scandir("/proc/%s/fd" % entry.name):
+                        _fd_target = os.readlink(_fd.path)
+                        if not str.startswith(_fd_target, "socket"):
+                            continue
+                        _socket = _fd_target.split(":[")[-1][:-1]
```
Comment (ethercflow) [inline measurement/process/meta.py]: If it's a broken socket, eg `3: broken symbolic link to socket:[418388778]` , what should we do?

File: measurement/process/meta.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+from measurement.files import fileutils
+
+
+def find_process_by_port(port=None):
+    process_list = []
+    if not port:
+        logging.fatal("No process listening port specified.")
+        return
+
+    # literate over all file descriptors and build a socket address -> pid map
```
Comment (LinuxGit) [inline measurement/process/meta.py]: s/literate/iterate

File: measurement/process/meta.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+                        except KeyError:
+                            result[_socket] = [entry.name]
+                except PermissionError:
+                    logging.warn(
+                        "Permission Denied reading /proc/%s/fd" % entry.name)
+        return result
+
+    def find_inode_by_port(port):
+        result = set()
+        netstat_files = ["/proc/net/tcp",
```
Comment (LinuxGit) [inline measurement/process/meta.py]: maybe we need to ignore udp or differentiate tcp/udp by argument

File: measurement/process/meta.py (hunk 2 above)
Comment (AstroProfundis) [reply measurement/process/meta.py]: It's trying to find the PID that holds a given listening port, so if the socket is broken, we're safe to ignore here.

File: measurement/process/meta.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            listen_list += fileutils.read_file(netstat_file).split("\n")
+        for line in listen_list:
+            if not line or "local_address" in line:
+                continue
+            _parts = line.split()
+            _local_addr = _parts[1]
+            _socket_st = _parts[3]
+            _inode_addr = _parts[9]
+            _local_port = int(_local_addr.split(":")[1], 16)
+            # st of '0A' is TCP_LISTEN, and '07' is for UDP
```
Comment (LinuxGit) [inline measurement/process/meta.py]: add comment: https://github.com/torvalds/linux/blob/master/include/net/tcp_states.h#L16

I don't find UDP states enum, but udp port 123  st is '07' when I started ntpd service.

Comment (AstroProfundis) [reply measurement/process/meta.py]: st of `07` is `TCP_CLOSE` in `tcp_states.h`, and all UDP ports shows as `07` on my machine. I think it's reasonable as there is no "connection" for UDP.

**Pull request merged**